packages/jest-config/src/normalize.js

```diff
@@ -266,13 +266,33 @@ const normalizePreprocessor = (options: InitialOptions): InitialOptions => {
 };
 
 const normalizeMissingOptions = (options: InitialOptions): InitialOptions => {
+  const knownRootDirs = {};
   if (!options.name) {
     options.name = crypto
       .createHash('md5')
       .update(options.rootDir)
       .digest('hex');
   }
 
+  if (Array.isArray(options.projects)) {
+    options.projects = options.projects.map((project, index) => {
```

Comment:
`normalize` is run on every project config, once per run, data is not persisted. I don't think it's necessary to double loop like this. Seems like simply adding a timestamp to the equation fixes the problem (or maybe I'm missing a bigger picture @SimenB):

```diff
diff --git a/packages/jest-config/src/normalize.js b/packages/jest-config/src/normalize.js
index 713c8b0da..11d61e080 100644
--- a/packages/jest-config/src/normalize.js
+++ b/packages/jest-config/src/normalize.js
@@ -270,6 +270,7 @@ const normalizeMissingOptions = (options: InitialOptions): InitialOptions => {
     options.name = crypto
       .createHash('md5')
       .update(options.rootDir)
+      .update(Date.now().toString())
       .digest('hex');
   }
```

Reply:
`Math.random()` seems safer in case the loop completes in less than a 1ms, but I agree that _should_ probably solve it as well

Reply:
Agreed. I'd go with `uuid/v4` for generating random number though, as `Math.random` is not so random after all and I'd hate to debug issue like this 😅.
Tested it locally and it fixes the issue (on Node 8, this particular example is not reproducible on Node 10)

Reply:
@thymikee which double loop do you mean? It's just a single loop through `projects`, isn't it?

Reply:
>@thymikee which double loop do you mean? It's just a single loop through projects, isn't it?

`normalize` is run for every project config, so essentially we loop through `projects` once again here, hence double loop. And the solution I proposed doesn't have this issue.